File: Baekjoon/Stack/10828.py
# https://www.acmicpc.net/problem/10828\

import sys

Stack = []
N = int(sys.stdin.readline())

for i in range(N):
    order = sys.stdin.readline().split()
    if order[0] == "push":
        Stack.append(order[1])
    elif order[0] == "pop":
        if len(Stack) > 0:
            print(Stack[-1])
            del Stack[-1]
        else:
            print("-1")
    elif order[0] == "size": 
        print(len(Stack))
    elif order[0] == "empty":
        if len(Stack) > 0:
            print("0")
        else:
            print("1")
    elif order[0] == "top":
        if len(Stack) > 0:
            print(Stack[-1])
        else:
            print("-1")

File: Baekjoon/Stack/10828.cpp
// https://www.acmicpc.net/problem/10828

#include <iostream>
#include <stack>
using namespace std;

int main() {
    int n;
    cin >> n;
    stack<int> s;
    while (n--) {
        string cmd;
        cin >> cmd;
        if (cmd == "push") {
            int num;
            cin >> num;
            s.push(num);
        } else if (cmd == "pop") {
            if (s.empty()) cout << -1 << '\n';
            else {
                cout << s.top() << '\n';
                s.pop();
            }
        } else if (cmd == "size") {
            cout << s.size() << '\n';
        } else if (cmd == "empty") {
            cout << s.empty() << '\n';
        } else if (cmd == "top") {
            if (s.empty()) cout << -1 << '\n';
            else cout << s.top() << '\n';
        }
    }
    return 0;
}